### PR TITLE
Send email when added to team

### PR DESCRIPTION
**Pull request by marcgreenstock**
This addresses the first item on https://github.com/rails-girls-summer-of-code/rgsoc-teams/issues/144 (Send an email when added to a team).

A few notes on what I have done:

1. I have monkey patched Mail::Message to respond to `:deliver_later` in `config/initializers/monkey_patch_mail.rb`. I've done this to make upgrading to Rails 4.2 easier when we have ActiveJob available. It essentially puts the mailer in a background job. This should be removed when upgrading to 4.2 .
2. I've added the [foreman gem](https://github.com/ddollar/foreman). It is nicer and easier to have your environment variables in a `.env` file. I've also added `.env` to the .gitignore file. I can update the readme to reflect this in the Mailtrap section.
3. I had a bit of trouble with other tests failing because they were trying to deliver the email. I've set `action_mailer.raise_delivery_errors` to `false`. Let me know if this is unacceptable, and I'll find another way.

Wiggle

Bed time!

### CODE REVIEW

**Comment by marcgreenstock** [comment]
Oh, one more thing:

I took advantage of [Action Mailer Previews](http://guides.rubyonrails.org/4_1_release_notes.html#action-mailer-previews). In development you can go to http://localhost:3000/rails/mailers and see a preview of the emails.

File: spec/mailers/role_mailer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require "spec_helper"
+
+describe RoleMailer, type: :mailer do
+  describe 'user_added_to_team' do
+    let(:user) { FactoryGirl.create(:user) }
+    let(:team) { FactoryGirl.create(:team) }
+    let(:role) { FactoryGirl.create(:mentor_role, user: user, team: team) }
+
+    before do
```
Comment (carpodaster) [inline spec/mailers/role_mailer_spec.rb]: I don't think we need to explicitly call subject before

Comment (carpodaster) [inline spec/mailers/role_mailer_spec.rb]: Nevermind, I just saw that you override subject to run the action :+1:

**Comment by carpodaster** [comment]
Thanks @marcgreenstock ! So many goodies in this PR, I can't even :smile: 

Two minor things: 

1. Can you add your `.env` as `.env-example` to the repo? I will take that one as base for #124 then

2. About `action_mailer.raise_delivery_errors = false`: it's odd that we get errors. I'll check it out locally myself to see what kind of errors we get. The tests should never send real mail, but maybe there are some expectations failing due to mails being shoved into background queue.

**Comment by marcgreenstock** [comment]
Hi @carpodaster just added to this RP. i like to use [letter_opener](https://github.com/ryanb/letter_opener) to open emails on development. Let me know if this is something you would like to use as a fallback to mailtrap.

**Comment by anikalindtner** [comment]
thanks so much @marcgreenstock!!!! :sparkles:

**Comment by marcgreenstock** [comment]
@carpodaster  regarding `raise_delivery_errors`, I'm thinking it is because of `require 'sucker_punch/testing/inline'` on [spec_helper.rb#L32](https://github.com/rails-girls-summer-of-code/rgsoc-teams/blob/master/spec/spec_helper.rb#L32). :)

**Comment by carpodaster** [comment]
@marcgreenstock  I could make the tests pass with this when `raise_delivery_errors` was set back to true:

```diff
--- a/app/mailers/role_mailer.rb
+++ b/app/mailers/role_mailer.rb
@@ -6,2 +6,3 @@ class RoleMailer < ActionMailer::Base
   def user_added_to_team(role)
+    return unless role.user.email.present?
     @role = role
```
It's a bit too defensive for my taste, but technically we can't guarantee that every user has an email address. `User#complete_from_github` should grab it, but only if they entered a legal GH handle.

**Comment by marcgreenstock** [comment]
@carpodaster no dramas, I've set it back to default in the test environment and added a robust check on the mailer. I think this is good practice anyway.

**Comment by carpodaster** [comment]
@marcgreenstock Superawesome – ¡muchas gracias!  Let's :shipit:

**Comment by alicetragedy** [comment]
:heart_eyes: 
also, DAT GIF

**Pull request merged**